docs/madr/decisions/076-naming-internal-envoy-resources.md

```diff
@@ -0,0 +1,256 @@
+
+### As a Kuma developer I want to distinguish between system and user resources in Envoy
+
+### As a Kuma developer I want to distinguish between system resources exposed and not exposed in metrics
+
+### As a user I want the cardinality of these resource names to be low
+
+Need to avoid:
+- IPs / non constant bits
+- Avoid dataplane name or kri.
+
+## Non use cases
+
+### As a user I want my MeshProxyPatch to continue working after the change
+
+This is not possible.
+In MeshProxyPatch you can use `name` matcher to match a resource name but if we change the name of the resource, it will not match anymore.
+The feature will be behind a feature flag so that nothing breaks unexpectedly but migration of MeshProxyPatch resources will be required.
+
+### As a Kuma developer I want to rename resources that are coming from outside Kuma
+
+This is either not possible or very hard.
+Envoy can connect to many xDS servers to get configuration from.
+If we integrate with systems like Spire (that exposes its own SDS server) we will not be able to rename resources that are coming from there.
+
+See:
+- https://github.com/spiffe/spire/blob/f687bf21e812a4bf027d88136032fc46497e7fe0/pkg/agent/endpoints/sdsv3/handler.go#L395
+- https://github.com/spiffe/spire/blob/f6a11a0ae03353b3444007ecaa8c81b34931bb49/pkg/agent/endpoints/sdsv3/handler.go#L282
+- https://github.com/spiffe/spire/blob/f6a11a0ae03353b3444007ecaa8c81b34931bb49/pkg/agent/endpoints/sdsv3/handler.go#L291
+- https://github.com/spiffe/spire/blob/f6a11a0ae03353b3444007ecaa8c81b34931bb49/pkg/agent/endpoints/sdsv3/handler.go#L312
+
+## Design
+
+### Use a `^system_([a-z0-9-]*_?)+$` regex to name system resources
+
+All changes will be behind the same feature flag as in [Migrating to consistent and well-defined naming for non-system Envoy resources and stats](./077-migrating-to-consistent-and-well-defined-naming-for-non-system-envoy-resources-and-stats.md).
+
+All system resources will conform to `^system_([a-z0-9-]*_?)+$` regex (we shouldn't use `system:` because of [this issue with `:` as separator](https://github.com/kumahq/kuma/issues/2363)).
+
+System resources that can be traced back to a Kuma resource with a valid KRI have the format `system_<KRI>`.
+
+For example:
+
+```
+system_kri_mtr_mesh-1__kuma-system_my-meshtrace_
+```
+
+Which indicates a system resource that was created for a `MeshTrace` resource.
+
+Resources that are not correlated with any Kuma resource like:
+
+```
+kuma:envoy:admin
+```
+
+Will conform with the previously mentioned regex:
+
+```
+system_envoy_admin
+```
+
+and should describe the resource as accurately and plainly as possible (it **MUST** take into account any related configuration option or annotation if exists).
+
+It means that `listener.0.0.0.0_15001` which can be configured by (`kuma.io/transparent-proxying-outbound-port`) will become `system_transparent_proxy_outbound_listener`.
+
```

Comment:
IMO this MADR is still a good place to define `self_` WDYT?

Reply:
`self_` is defined here - https://github.com/kumahq/kuma/pull/13485/files#diff-15d3fac527452e5002d13b52e9b646b84ed3df20db13fcbd2ed0ed3af3bc6355R29. You mentioned [here](https://github.com/kumahq/kuma/pull/13783#discussion_r2161454561) - "In other words things that are not really a result of the environment so things not coming from Mesh*Service Mesh*Route". Pods that are then converted to DPs are definitely coming from the environment.